web/frontend/src/pages/election/components/SubjectComponent.tsx

```diff
@@ -0,0 +1,271 @@
+        newRanks[newRanks.findIndex((rank) => rank.ID === targetID)] = targetObject;
+        setSubject({ ...subject, Ranks: newRanks });
+        break;
+      case 'SELECT':
+        const newSelects = [...Selects];
+        newSelects[newSelects.findIndex((select) => select.ID === targetID)] = targetObject;
+        setSubject({ ...subject, Selects: newSelects });
+        break;
+      case 'TEXT':
+        const newTexts = [...Texts];
+        newTexts[newTexts.findIndex((text) => text.ID === targetID)] = targetObject;
+        setSubject({ ...subject, Texts: newTexts });
+        break;
+      default:
+        break;
+    }
+  };
+
+  const addSubject = () => {
+    const newSubjects = [...Subjects];
+    const newSubj = newSubject();
+    newSubjects.push(newSubj);
+    setSubject({ ...subject, Subjects: newSubjects, Order: [...Order, newSubj.ID] });
+  };
+
+  const addQuestion = (type: 'RANK' | 'TEXT' | 'SELECT') => {
+    switch (type) {
+      case 'RANK':
+        const newRanks = [...Ranks];
+        const rank = newRank();
+        newRanks.push(rank);
+        setSubject({ ...subject, Ranks: newRanks, Order: [...Order, rank.ID] });
+        break;
+      case 'TEXT':
+        const newTexts = [...Texts];
+        const text = newText();
+        newTexts.push(text);
+        setSubject({ ...subject, Texts: newTexts, Order: [...Order, text.ID] });
+        break;
+      case 'SELECT':
+        const newSelects = [...Selects];
+        const select = newSelect();
+        newSelects.push(select);
+        setSubject({ ...subject, Selects: newSelects, Order: [...Order, select.ID] });
+        break;
+      default:
+        break;
+    }
+  };
+
+  const localRemoveSubject = (subjID: ID) => () => {
+    const newSubjects = [...Subjects];
+    setSubject({
+      ...subject,
+      Subjects: newSubjects.filter((subj) => subj.ID !== subjID),
+      Order: Order.filter((id) => id !== subjID),
+    });
+  };
+
+  const removeChildQuestion = (targetID: ID, type: 'RANK' | 'TEXT' | 'SELECT') => () => {
+    switch (type) {
+      case 'RANK':
+        setSubject({
+          ...subject,
+          Ranks: Ranks.filter((rank) => rank.ID !== targetID),
+          Order: Order.filter((id) => id !== targetID),
+        });
+        break;
+      case 'TEXT':
+        setSubject({
+          ...subject,
+          Texts: Texts.filter((text) => text.ID !== targetID),
+          Order: Order.filter((id) => id !== targetID),
+        });
+        break;
+      case 'SELECT':
+        setSubject({
+          ...subject,
+          Selects: Selects.filter((select) => select.ID !== targetID),
+          Order: Order.filter((id) => id !== targetID),
+        });
+        break;
+      default:
+        break;
+    }
+  };
+
+  // Sorts the questions components & sub-subjects according to their Order into
+  // the components state array
+  useEffect(() => {
+    const findSubject = (id: string) => {
+      if (Subjects) {
```

Comment:
Since the implementation of yup and the `require` method now this check is indeed useless, before (with only AJV) it could break things to not have this check.

Reply:
Not anymore now that we have Yup otherwise AJV would allow these fields to be undefined, indeed I'll delete the checking